### ISSUE TITLE
Commit All Changes button is missing via Netopeer-GUI when connected to a device

### ISSUE THREAD

**Issue by tbrito01** [CESNET/Netopeer-GUI#84]
Hi,
1. Changed Netopeer-GUI from Single Instance Mode successfully. Used step 8 from README file.  Ran into a problem while doing step 8 and was advised to [sqlite, which is used as DB, doesn't support alter table. run 'rm /var/www/netopeer-gui/app/netopeerWebGui.db' (described in step 7 in readme]
2. Connected to a remote server successfully. 
3.  "Commit All Changes button" is not displayed from any menu tab choices. 

Ran the demo GUI and see the button. However we need to connect to our own server to run some testing so the demo GUI won't work for us.

This is a bug on your GUI tool. Please advise.

**Comment by tbrito01** [comment]
One thing that I wonder is our modules require "commit config"  on a candidate datastore not Running datastore like your Demo GUI. Could that be an issue why the commit button does not show up?

**Comment by cejkato2** [comment]
Hello,
I'm sorry that I didn't answer to your issues earlier.
I went through your issues briefly and it seems to me that the result of #82 is similar to this issue.
I have few ideas:
1) @tbrito01 Does your NETCONF server provide proper data models in yang/yin via get-schema operation of NETCONF protocol?
2) @alexadavid Is it possible that NetopeerGUI doesn't show the button for saving changes in case a model is not available?
3) @alexadavid Does NetopeerGUI show the button when there is no configuration info in a current section/tab?
4) @tbrito01 Does your data model contain any configuration information? Can you provide us your data model? Is it possible to get your NETCONF server with your modules in order to test it in our environment? (maybe it would help us to get it running sooner...)
5) Is it possible to plan some chat/hangout together? It may be long discussion but still it can be a faster communication...
Tomas

**Comment by tbrito01** [comment]
Hi Thomas,

Thank you so very much for responding to my email about the button. Here are the responses to your questions addressed to me:

1) Our yang models are working with other tools as a test.  Do you think we should have to reformat our yang modules in yin? Does your tool require yin models or are they capable of working directly with yang models? (I believe that is is the normal case (yang).  Yin is only provided as a way to enable use of XML tools in the netconf development). Isn't it?
 While they are trying to find a solution, can we use the Netopeer CLI interface as an alternative.  This might be useful because it would let us know if the problems are just GUI issues or if there are fundamental issues.

4) Unfortanetly due to our company's policies we can't share any configuration information or data model or provide our NETCONF server with our modules without a Non Disclosure Agreement. I know it would help us in getting us to the problem faster, but maybe we can find another alternative. I really think the problem is with the GUI tool. If there were any problems with our configuration, then perhaps the button would be grayed out or disabled but not missing completely from the GUI. 

5) Absolutely, please provide time slots this week that would work for you and I will look at my schedule so we can set up a meeting.

I really appreciate your help on this Tomas. 

Tatiana

**Comment by alexadavid** [comment]
Hello,
sorry for my late answer.

2, 3) I went deeply through our code and I've found condition for showing commit all button based on device capability for
```
const CPBLT_WRITABLERUNNING		= "urn:ietf:params:netconf:capability:writable-running:1.0";
```
, which must be set for showing button. Otherwise commit all button is hidden - and this could be your problem. I'll discuss this problem and solutions with @cejkato2 today.

**Comment by cejkato2** [comment]
@tbrito01 Can you please send us NETCONF hello message? It is not a confidential information so I think, it is shareable...
The important thing for us is the list of capabilities.

You can find it as an icon on top:
![netopeergui-sessioninfo](https://cloud.githubusercontent.com/assets/205818/11057249/b58c082c-8789-11e5-86cb-f6a1f272f430.png)

**Comment by tbrito01** [comment]
Hi Thomas and David,

I sent you just now a Hello message via Session Info. Did you get it?

@David: Thank you for checking your code. You might have found the root
cause of this. :)

Please let me know if I can be of any further assistance.

Thank you,

Tatiana

**Comment by tbrito01** [comment]
Hi Thomas,

Here is the list of capabilities when sending a NETCONF hello message:

![capture](https://cloud.githubusercontent.com/assets/15365533/11063765/dce7cc10-8786-11e5-9382-dbe600669b7f.PNG)

It appears that David fixed the Commit All Changes button issue. How can I get the fix?

Thank you,

Tatiana

**Comment by alexadavid** [comment]
fixed in #85

**Comment by tbrito01** [comment]
Hi Tomas,

I had to reach out to you since you were very helpful last time. We got a
fix from David on the issue with the Commit All button not showing up in
the GUI when connected to a remote server. However it does not work. We
followed David's steps and still does not appear. We tried to reach out to
David several times but no answer.  I was wondering if you could reach
David and find out what could be wrong.

Thanks in advance,

Tatiana
